libs/core/langchain_core/language_models/chat_models.py

```diff
@@ -950,6 +958,202 @@ def bind_tools(
     ) -> Runnable[LanguageModelInput, BaseMessage]:
         raise NotImplementedError()
 
+    def with_structured_output(
+        self,
+        schema: Union[Dict, Type[BaseModel]],
+        *,
+        method: Literal["function_calling", "json_mode"] = "function_calling",
```

Comment:
dunno if this should be on base implementation

Reply:
dropped